File: test_get_product.py
import httpx
import json

def get_env_robust():
    env = {}
    with open(".env.local", "r", encoding="utf-8") as f:
        for line in f:
            line = line.strip()
            if not line or line.startswith("#"):
                continue
            if "=" in line:
                key, val = line.split("=", 1)
                env[key.strip()] = val.strip().strip('"')
    return env

def test():
    env = get_env_robust()
    domain = env.get("SHOPIFY_STORE_DOMAIN")
    token = env.get("SHOPIFY_STOREFRONT_ACCESS_TOKEN")
    
    url = f"https://{domain}/api/2025-04/graphql.json"
    headers = {
        "X-Shopify-Storefront-Access-Token": token,
        "Content-Type": "application/json"
    }
    
    query = """
    query getProductByHandle($handle: String!) {
      product(handle: $handle) {
        title
        handle
      }
    }
    """
    
    handle = "hb-body-εποξειδικό-αστάρι-σπρέι-γκρι-p981"
    
    print(f"Testing handle: {handle}")
    response = httpx.post(url, headers=headers, json={
        "query": query,
        "variables": {"handle": handle}
    })
    
    print("Response:")
    print(json.dumps(response.json(), indent=2, ensure_ascii=False))

if __name__ == "__main__":
    test()

File: test_get_product.js
const { getProduct } = require('./src/lib/shopify/client');

async function run() {
    try {
        const handle = "hb-body-εποξειδικό-αστάρι-σπρέι-γκρι-p981";
        console.log(`Testing getProduct with handle: "${handle}"`);
        const p = await getProduct(handle);
        if (p) {
            console.log("FOUND PRODUCT:", p.title);
        } else {
            console.log("PRODUCT NOT FOUND (returned undefined or null)");
        }
    } catch (e) {
        console.error("ERROR in getProduct:", e);
    }
}
run();
